### PR TITLE
[Interop] Fixing rados_utils.py: Fix for osd wait method

### PR DESCRIPTION
**Pull request by pdhiran**
Corrected the method that waits for 120 seconds for the OSD to go down.

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

### CODE REVIEW

**Comment by pdhiran** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614891674199/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614891674199/result.html

File: ceph/rados_utils.py [hunk 1]
```diff
@@ -221,9 +221,11 @@ def check_osd_up(self, osd_id):
         while end_time > datetime.datetime.now():
             status = self.is_up(osd_id)
             if status:
-                return 1
-            time.sleep(5)
-        return 0
+                time.sleep(5)
```
Comment (psathyan) [inline ceph/rados_utils.py:224]: The method returns true when the OSD is up... but this change returns true when it is down. We could fix the return code to True (or 0) when OSD is up. 

```
while end_time > datetime.datetime.now():
  if self.is_up(osd_id): return True
  time.sleep(5)

return False
```

Comment (psathyan) [reply ceph/rados_utils.py:224]: For interop, can we look at the code where the check is for down and negate at that location.

Comment (pdhiran) [reply ceph/rados_utils.py:224]: The method returns true when OSD is up. ( after waiting for 120 seconds ), and returns false as soon as OSD goes down. This is expected of the method. Do we want to change the functionality?

Comment (psathyan) [reply ceph/rados_utils.py:224]: My bad... sorry for the misdirect. I got confused with 1 being `False`. Do we want to wait for `120` seconds to return the OSD is up?

Comment (pdhiran) [reply ceph/rados_utils.py:224]: We wait for 120 seconds to return when OSD is down. We stay in the while loop until the said OSD is still up .

File: ceph/rados_utils.py [hunk 2]
```diff
@@ -219,10 +219,27 @@ def check_osd_up(self, osd_id):
...
+
+        Returns: 1 if up, 0 if down
+
+        """
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
+        while end_time > datetime.datetime.now():
+            time.sleep(5)
+            if self.is_up(osd_id):
+                print("OSD is up")
+                return 1
```
Comment (psathyan) [inline ceph/rados_utils.py:242]: IMHO, we can achieve this with a single method `def wait_until(self, osd, up=True) -> bool:` as the difference is the state of OSD daemon.  For example
```
def wait_until_osd_state(self, osd, down=True) -> bool:
  end_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
  while end_time > datetime.datetime.now():
    time.sleep(5)
     osd_is_up = self.is_up(osd_id):
     if osd_is_up and not down: return True
     if down and not osd_is_up: return True

  return False
```

Comment (pdhiran) [reply ceph/rados_utils.py]: Added a single method.

**Comment by pdhiran** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614927542498/

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614927542498/result.html

**Review by psathyan — APPROVED** [review]
Looks good to me

**Pull request merged**